core/src/scene/light.h

```diff
@@ -27,6 +27,15 @@ enum class LightOrigin {
     world
 };
 
+static std::string lightOriginString(LightOrigin origin) {
```

Comment:
You can make this a member function of the `LightOrigin` enum class. Something like `toString`.